libraries/Microsoft.Bot.Builder/ActivityHandler.cs

```diff
@@ -380,6 +383,86 @@ protected virtual Task OnEventAsync(ITurnContext<IEventActivity> turnContext, Ca
             return Task.CompletedTask;
         }
 
+        /// <summary>
+        /// Invoked when an invoke activity is received from the connector when the base behavior of
+        /// <see cref="OnTurnAsync(ITurnContext, CancellationToken)"/> is used.
+        /// Invoke activities can be used to communicate many different things.
+        /// By default, this method will call <see cref="OnSignInInvokeAsync(ITurnContext{IInvokeActivity}, CancellationToken)"/> if the
+        /// activity's name is <c>signin/verifyState</c> or <c>signin/tokenExchange</c>.
+        /// A <c>signin/verifyState</c> or <c>signin/tokenExchange</c> invoke can be triggered by an <see cref="OAuthCard"/>.
+        /// </summary>
+        /// <param name="turnContext">A strongly-typed context object for this turn.</param>
+        /// <param name="cancellationToken">A cancellation token that can be used by other objects
+        /// or threads to receive notice of cancellation.</param>
+        /// <returns>A task that represents the work queued to execute.</returns>
+        /// <remarks>
+        /// When the <see cref="OnTurnAsync(ITurnContext, CancellationToken)"/>
+        /// method receives an invoke activity, it calls this method.
+        /// If the event <see cref="IInvokeActivity.Name"/> is `signin/verifyState` or `signin/tokenExchange`, it calls
+        /// <see cref="OnSignInInvokeAsync(ITurnContext{IInvokeActivity}, CancellationToken)"/>
+        /// Invoke activities communicate programmatic commands from a client or channel to a bot.
+        /// The meaning of an invoke activity is defined by the <see cref="IInvokeActivity.Name"/> property,
+        /// which is meaningful within the scope of a channel.
+        /// A `signin/verifyState` or `signin/tokenExchange` invoke can be triggered by an <see cref="OAuthCard"/> or an OAuth prompt.
+        /// </remarks>
+        /// <seealso cref="OnTurnAsync(ITurnContext, CancellationToken)"/>
+        protected virtual Task OnInvokeActivityAsync(ITurnContext<IInvokeActivity> turnContext, CancellationToken cancellationToken)
```

Comment:
>OnInvokeActivityAsync [](start = 31, length = 21)

shouldn't this and other invoke activity handlers return an InvokeResponse?

Reply:
No, the pattern is for calls to send an InvokeResponse activity from wherever you are in your dialog code...bubbling all the way back up would be hard given all the other API signatures of not returning this.